components/eam/src/physics/crm/cloud_diagnostics.F90

```diff
@@ -1,525 +1,374 @@
+  ! call outfld('rel_cloud', rel, pcols, lchnk)
+  ! call outfld('rei_cloud', rei, pcols, lchnk)
+
+  call pbuf_get_field(pbuf, cicewp_idx, cicewp )
+  call pbuf_get_field(pbuf, cliqwp_idx, cliqwp )
+
+  ! Compute liquid and ice water paths
+
+  ! ----------------------------------------------------------- !
+  ! Adjust in-cloud water values to take account of convective  !
+  ! in-cloud water. It is used to calculate the values of       !
+  ! iclwp and iciwp to pass to the radiation.                   !
+  ! ----------------------------------------------------------- !
+  if( conv_water_in_rad /= 0 ) then
+    allcld_ice(:ncol,:) = 0._r8 ! Grid-avg all cloud liquid
+    allcld_liq(:ncol,:) = 0._r8 ! Grid-avg all cloud ice
+
+    call conv_water_4rad( state, pbuf, conv_water_in_rad, allcld_liq, allcld_ice )
+  else
+    allcld_liq(:ncol,top_lev:pver) = state%q(:ncol,top_lev:pver,ixcldliq)  ! Grid-ave all cloud liquid
+    allcld_ice(:ncol,top_lev:pver) = state%q(:ncol,top_lev:pver,ixcldice)  !           "        ice
+  end if
+
+  ! ------------------------------------------------------------ !
+  ! Compute in cloud ice and liquid mixing ratios                !
+  ! Note that 'iclwp, iciwp' are used for radiation computation. !
+  ! ------------------------------------------------------------ !
+  do k = 1,pver
+    do i = 1,ncol
+      iciwp(i,k)  = 0._r8
+      iclwp(i,k)  = 0._r8
+      ! icimr(i,k)  = 0._r8
+      ! icwmr(i,k)  = 0._r8
+      ! iwc(i,k)    = 0._r8
+      ! lwc(i,k)    = 0._r8
+      gicewp(i,k) = 0._r8
+      gliqwp(i,k) = 0._r8
+      cicewp(i,k) = 0._r8
+      cliqwp(i,k) = 0._r8
+    end do
+  end do
+
+  ! do k = top_lev, pver
+  !   do i = 1, ncol
+  !     ! Limits for in-cloud mixing ratios consistent with MG microphysics
+  !     ! in-cloud mixing ratio maximum limit of 0.005 kg/kg
+  !     icimr(i,k)     = min( allcld_ice(i,k) / max(0.0001_r8,cld(i,k)),0.005_r8 )
+  !     icwmr(i,k)     = min( allcld_liq(i,k) / max(0.0001_r8,cld(i,k)),0.005_r8 )
+  !     iwc(i,k)       = allcld_ice(i,k) * state%pmid(i,k) / (287.15_r8*state%t(i,k))
+  !     lwc(i,k)       = allcld_liq(i,k) * state%pmid(i,k) / (287.15_r8*state%t(i,k))
+  !     ! Calculate total cloud water paths in each layer
+  !     iciwp(i,k)     = icimr(i,k) * state%pdel(i,k) / gravit
+  !     iclwp(i,k)     = icwmr(i,k) * state%pdel(i,k) / gravit
+  !   end do
+  ! end do
+
+  ! do k = 1,pver
+  !   do i = 1,ncol
+  !     gicewp(i,k) = iciwp(i,k)*cld(i,k)
+  !     gliqwp(i,k) = iclwp(i,k)*cld(i,k)
+  !     cicewp(i,k) = iciwp(i,k)
+  !     cliqwp(i,k) = iclwp(i,k)
+  !   end do
+  ! end do
+
+  ! Determine parameters for maximum/random overlap
+  call cldovrlap(lchnk, ncol, state%pint, cld, nmxrgn, pmxrgn)
+
+  ! Cloud cover diagnostics
+  call cloud_cover_diags_out(lchnk, ncol, cld, state%pmid, nmxrgn, pmxrgn )
+  
+  tgicewp(:ncol) = 0._r8
+  tgliqwp(:ncol) = 0._r8
+
+  do k=1,pver
+    tgicewp(:ncol)  = tgicewp(:ncol) + gicewp(:ncol,k)
```

Comment:
It looks like gicewp is never set. The code block that sets these has been commented out above, so it looks like gicewp and gliqwp are all zeros here.

Reply:
see other comment - this will be deleted in upcoming commits since this calculation is done elsewhere for the MMF